satpy/readers/hdf5_utils.py

```diff
@@ -120,3 +122,11 @@ def get(self, item, default=None):
             return self[item]
         else:
             return default
+
+
+def from_h5_array(h5dset):
+    """Create a dask array from an h5py dataset, ensuring uniqueness of the dask array name."""
+    name_str = os.fspath(h5dset.file.filename) + "-" + h5dset.name
+    name = hashlib.md5(name_str.encode(), usedforsecurity=False).hexdigest()
+    dset_data = da.from_array(h5dset, chunks=CHUNK_SIZE, name=name)
+    return dset_data
```

Comment:
I think because you are providing your own name you must also include the dtype and chunk size and possibly the shape of the array in your name's generation. Maybe this isn't a problem realistically since you are using the `.name` of the `h5dset` (which I assume is the variable name?), but if someone loaded the same variable with a different chunk size I *think* the dask array's `.name` would be the same and dask would get confused (using chunks from one chunk size in place of the other array with a different chunk size). Again, not sure how possible that is in Satpy, but maybe there should at least be a warning in the docstring if you don't include the chunk size in the `name` generation.

Reply:
ah, indeed, good call! I will change that to include the chunk size. As for dtype, I don't think it's necessary though, as we don't pass that to from_array.

Reply:
Ok, this is now addressed, tell me what you think!